fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb

```diff
@@ -2,9 +2,64 @@ module Fastlane
   module Actions
     class UploadSymbolsToCrashlyticsAction < Action
       def self.run(params)
-        params[:binary_path] ||= (Dir["/Applications/Fabric.app/**/upload-symbols"] + Dir["./Pods/**/upload-symbols"]).last
-        UI.user_error!("Please provide a path to the binary using binary_path:") unless params[:binary_path]
+        require 'tmpdir'
+
+        find_binary_path(params)
+        find_api_token(params)
+
+        dsym_paths = []
+        dsym_paths << params[:dsym_path]
+        dsym_paths += Actions.lane_context[SharedValues::DSYM_PATHS] if Actions.lane_context[SharedValues::DSYM_PATHS]
+        UI.error("Couldn't find any dSYMs, please pass them using the dsym_path option") if dsym_paths.count == 0
+
+        # Get rid of duplicates (which might occur when both passed and detected)
+        dsym_paths = dsym_paths.collect { |a| File.expand_path(a) }
+        dsym_paths.uniq!
 
+        dsym_paths.each do |current_path|
+          handle_dsym(params, current_path)
+        end
+
+        UI.success("Successfully uploaded dSYM files to Crashlytics 💯")
+      end
+
+      # @param current_path this is a path to either a dSYM or a zipped dSYM
+      #   this might also be either nested or not, we're flexible
+      def self.handle_dsym(params, current_path)
+        if current_path.end_with?(".dSYM")
+          upload_dsym(params, current_path)
+        elsif current_path.end_with?(".zip")
+          UI.message("Extracting '#{current_path}'...")
+
+          current_path = File.expand_path(current_path)
+          Dir.mktmpdir do |dir|
+            Dir.chdir(dir) do
+              Actions.sh("unzip -qo #{current_path.shellescape}")
+              Dir["*.dSYM"].each do |sub|
+                handle_dsym(params, sub)
+              end
+            end
+          end
+        else
+          UI.error "Don't know how to handle '#{current_path}'"
+        end
+      end
+
+      def self.upload_dsym(params, path)
+        UI.message("Uploading '#{path}'...")
+        command = []
+        command << File.expand_path(params[:binary_path])
```

Comment:
Has this already been expanded at https://github.com/fastlane/fastlane/pull/4017/files#diff-2eec78d70c41940712d3172fc80d1898R78?